middleware/rpc/provider.go

```diff
@@ -38,7 +38,7 @@ import (
 // It is intended that Provider is used to wrap legacy native provider implementations
 // while they are gradually transferred over to pulumi-go-provider based implementations.
 //
-// Construct, Call and StreamInvoke are not supported and will always return
+// Call and StreamInvoke are not supported and will always return
```

Comment:
StreamInvoke is deprecated now, so we can probably remove that from the comment too. I wonder how hard it is to add `Call` now we've made `Construct` work 🤔

Reply:
I do believe that Call should be reworked to use the Host interface, and not have a callback in the CallRequest struct. I intend to do that as a follow-up.